telos-frontend/src/components/calendar/JournalCalendar.jsx

```diff
@@ -1,162 +1,153 @@
+    super(props);
+    this.state = {
+      data: calendarData,
+      resources: [
+        {
+          fieldName: 'priorityId',
+          title: 'Priority',
+          instances: priorityData,
+          allowMultiple: false,
+        },
+        {
+          fieldName: 'typeId',
+          title: 'Type',
+          instances: typeData,
+          allowMultiple: false,
+        },
+      ],
+    };
 
-const reducer = (state, action) => {
-  switch (action.type) {
-    case 'setLoading':
-      return { ...state, loading: action.payload };
-    case 'setData':
-      return { ...state, data: action.payload.map(mapAppointmentData) };
-    case 'setCurrentViewName':
-      return { ...state, currentViewName: action.payload };
-    case 'setCurrentDate':
-      return { ...state, currentDate: action.payload };
-    default:
-      return state;
+    this.commitChanges = this.commitChanges.bind(this);
   }
-};
 
-export default () => {
-  const [state, dispatch] = React.useReducer(reducer, initialState);
-  const { data, loading, currentViewName, currentDate } = state;
-  const setCurrentViewName = React.useCallback(
-    (nextViewName) =>
-      dispatch({
-        type: 'setCurrentViewName',
-        payload: nextViewName,
-      }),
-    [dispatch]
-  );
-  const setData = React.useCallback(
-    (nextData) =>
-      dispatch({
-        type: 'setData',
-        payload: nextData,
-      }),
-    [dispatch]
-  );
-  const setCurrentDate = React.useCallback(
-    (nextDate) =>
-      dispatch({
-        type: 'setCurrentDate',
-        payload: nextDate,
-      }),
-    [dispatch]
-  );
-  const setLoading = React.useCallback(
-    (nextLoading) =>
-      dispatch({
-        type: 'setLoading',
-        payload: nextLoading,
-      }),
-    [dispatch]
-  );
+  commitChanges({ added, changed, deleted }) {
+    this.setState((state) => {
+      let { data } = state;
+      if (added) {
+        const startingAddedId = data.length > 0 ? data[data.length - 1].id + 1 : 0;
+        data = [...data, { id: startingAddedId, ...added }];
+      }
+      if (changed) {
+        data = data.map((appointment) =>
+          changed[appointment.id] ? { ...appointment, ...changed[appointment.id] } : appointment
+        );
+      }
+      if (deleted !== undefined) {
+        data = data.filter((appointment) => appointment.id !== deleted);
+      }
+      return { data };
+    });
+  }
 
-  React.useEffect(() => {
-    getData(setData, setLoading);
-  }, [setData, currentViewName, currentDate]);
+  render() {
+    const { data, resources } = this.state;
 
-  return (
-    <Paper style={{ height: '100%' }}>
-      <ThemeProvider theme={theme}>
-        <Scheduler data={data} height="full">
-          <ViewState
-            currentDate={currentDate}
-            currentViewName={currentViewName}
-            onCurrentViewNameChange={setCurrentViewName}
-            onCurrentDateChange={setCurrentDate}
-          />
+    return (
+      <Paper>
```

Comment:
Can you keep the height styling here? Otherwise the widget goes beyond the bottom of the journal page

![image](https://user-images.githubusercontent.com/54158903/111900875-cb472980-8a99-11eb-99c0-d267b98ae1cf.png)

Reply:
Fixed now :)

Reply:
Thanks, looks good now